### DISCUSSION TITLE
Link style customization

### DISCUSSION THREAD

**Discussion by ivokst92** [clientIO/joint#1771] [Q&A] [answered]
Hello,
Is there any way to make different style of the link between the different vertices?
For example:
![image](https://user-images.githubusercontent.com/26894825/183631198-c85dea29-fd86-4fa1-82de-a8ff99cd11d4.png)
I want to have dashed line with different width between particular vertices.

**Comment by kumilingus** [comment, marked as answer]
You need to define a link with an extra *SVGPath* (or *SVGPolyline*) and then:

1. Set the points of the `SVGPolyline` whenever `vertices` changes.
Example: https://jsfiddle.net/kumilingus/5npeku14/
*Note: Styling is limited as the polyline is rendered on top of the original path*

2. Define a custom SVG Attribute
Example: https://jsfiddle.net/kumilingus/378e9vcL/
*Note: using an internal mechanism to define an SVG Attributes
 Available also via decorators - https://changelog.jointjs.com/rappid/docs/decorators#SVGAttribute*

**Comment by ivokst92** [comment]
Thank you!

Is there any way to use the SVG Attribute approach without appyling it between every vertices, but only between particular, for example:
![image](https://user-images.githubusercontent.com/26894825/184131743-458e6bbd-f0ac-4a46-9b50-6abddcbdbcf6.png)

**Reply by kumilingus** [reply]
You have to change the logic of `connectionPart`  attribute in this [demo](https://jsfiddle.net/kumilingus/378e9vcL/) not to look at all `vertices` but just a subset.

**Reply by ivokst92** [reply]
> You have to change the logic of `connectionPart` attribute in this [demo](https://jsfiddle.net/kumilingus/378e9vcL/) not to look at all `vertices` but just a subset.

Since I'm using typescript to achieve this, I'm not able to access "this" (the current instance) inside the static attributes

```js
static attributes = {
        connectionPart: {
            set: function(type) {
                const vertices =  <this>.model.vertices();
                if (type === 'inner') {
                    const path = new g.Path(new g.Polyline(vertices));
                    return { d: path.serialize() };
                } else {
                    if (vertices.length < 2) {
                        return { d: <this>.getSerializedConnection() };
                    } else {
                        const { sourcePoint: s, targetPoint: t } = this;
                        const v1 = vertices[0];
                        const v2 = vertices[vertices.length - 1];
                        return { d: `M ${s.x} ${s.y} ${v1.x} ${v1.y} M ${v2.x} ${v2.y} ${t.x} ${t.y}` };
                    }
                }
            }
        }
    };
```
Removing the "static" does nothing.
Any suggestions on how to workaround this?

**Reply by kumilingus** [reply]
You have to ignore the error with `// @ts-ignore` for now.

We should be passing the `cellView` instance as the last argument to the `set()` method (and others). Feel free to create a PR.
- https://github.com/clientIO/joint/blob/master/src/dia/CellView.mjs#L610
- https://github.com/clientIO/joint/blob/master/src/dia/CellView.mjs#L652
- https://github.com/clientIO/joint/blob/master/src/dia/CellView.mjs#L675
- https://github.com/clientIO/joint/blob/master/src/dia/CellView.mjs#L550

**Reply by ivokst92** [reply]
Ignoring the error didn't worked as expected, the code compiled successfully but at runtime "this" wasn't returning the instance of the cellView/link.

A PR was created.

**Reply by kumilingus** [reply]
The context of the `set()` method is a `cellView` (see [code](https://github.com/clientIO/joint/blob/master/src/dia/CellView.mjs#L610)).

1. please make to use a valid syntax (I don't think this is correct `<this>.getSerializedConnection()`)
2. do not use fat arrow function to define `set()`

**Reply by ivokst92** [reply]
Using the regular function instead of fat arrow function did the trick! 
I'll ignore the error until the next release.

Thank you!
